howso/train.amlg

```diff
@@ -227,6 +227,44 @@
 			)
 		)
 
+		;if dataset has features that are for derivation output only and are not being explicitly trained on
+		;then add them to the training data with nulls
+		(if (< (size features) (size !trainedFeatures))
```

Comment:
While this check is fast, I think it needs to be more proper such as `(size (remove (zip features) !trainedFeatures) )`.

This happens before !trainedFeatures is updated below. So here `features` could be a list of completely new features of the same number or more as !trainedFeatures.

Reply:
good point